autorest/azure/azure.go

```diff
@@ -29,12 +30,20 @@ const (
 
 // ServiceError encapsulates the error response from an Azure service.
 type ServiceError struct {
-	Code    string `json:"code"`
-	Message string `json:"message"`
+	Code    string         `json:"code"`
+	Message string         `json:"message"`
+	Details *[]interface{} `json:"details"`
 }
 
 func (se ServiceError) Error() string {
-	return fmt.Sprintf("Azure Error: Code=%q Message=%q", se.Code, se.Message)
+	if se.Details != nil {
+		d, err := json.Marshal(*(se.Details))
+		if err != nil {
+			return fmt.Sprintf("Code=%q Message=%q Details=%v", se.Code, se.Message, *se.Details)
+		}
+		return fmt.Sprintf("Code=%q Message=%q Details=%v", se.Code, se.Message, string(d))
+	}
```

Comment:
@ahmetalpbalkan I added a nil check for details. 
I use json marshal for converting details to string because if I use it directly, its printing with type..
```Go
fmt.Printf("Details=%s\n", *(r.ServiceError.Details))
// Details=[map[message:error message code:Conflict] map[code:Conflict2 message:error message]]
```
Marshaler prints it..
```Go
d, err := json.Marshal(*(se.Details))
fmt.Printf("Details=%v", string(d))
// Details=[{"code":"conflict1","message":"error message1"}]
```
Else if we don't want to use Marshal, I need to unwrap it somewhat like below. 
```Go
var s []string
	for _, v := range *(r.ServiceError.Details) {
		m, _ := v.(map[string]interface{})
		for k, val := range m {
			s = append(s, k+":"+val.(string))
		}
	}
	fmt.Println("Details=", s)

// Details= [code:Conflict message:error message code:Conflict2 message:error message]
```
What you think?